### PR TITLE
[lang-test] Factorize parsing logic for tests

### PR DESCRIPTION
**Pull request by oowekyala**
* Factorize logic to parse strings/resources files into a language-independent base class
* Remove a lot of ad-hoc implementations duplicated across all language modules
* This not only makes it much more straightforward to write new tests, it also puts the logic of applying additional visitors (symbol table and the like) into a single place. This will make it easier to refactor it to use processing stages on the 7.0 branch.
* This PR contains only changes for pmd-java and pmd-modelica. This makes the diff shorter, but hopefully representative enough to review the general approach. For the rest of the modules we can revert f72810088d after merging. They're more or less exactly like those.

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">This changeset introduces 0 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/b9d569651e0349d9972fcb568efca465/diff/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-modelica/src/test/java/net/sourceforge/pmd/lang/modelica/ModelicaParsingHelper.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.modelica;
+
+import org.jetbrains.annotations.NotNull;
```
Comment (adangel) [inline pmd-modelica/src/test/java/net/sourceforge/pmd/lang/modelica/ModelicaParsingHelper.java]: that looks strange here.... jetbrains notnull

Comment (oowekyala) [reply pmd-modelica/src/test/java/net/sourceforge/pmd/lang/modelica/ModelicaParsingHelper.java]: Oh yeah Intellij sometimes inserts this out of nowhere... it's a pain

**Comment by oowekyala** [comment]
> Do I see it right, that apex first needs to solve #1937 , so that we can use BaseParserHelper?

I don't remember why I excluded Apex... I don't think I paid attention to #1937. Maybe I just forgot. I think we should not wait for #1937, and just use RootNode as the type parameter for now. Doing it ASAP reduces the number of conflicts later on. I'll do that on master

> This is quite a lot of changes - I hope you know what you can expect in the PMD 7 branches... (I'll try the merge and let's see how many conflicts we get...)

Thanks for doing this, I'll try merging back to java-grammar later today if you don't

> Another point that came into mind: now it seems, we proposed a common way for language implementations how to deal with unit tests - I think, we should add this to https://pmd.github.io/latest/pmd_devdocs_major_adding_new_language.html , wdyt?

That's a good idea!

**Pull request merged**